packages/lexical-editor/src/plugins/TypographyPlugin/TypographyPlugin.tsx

```diff
@@ -16,8 +12,8 @@ export const TypographyPlugin: React.FC = () => {
             ADD_TYPOGRAPHY_ELEMENT_COMMAND,
             payload => {
                 const selection = $getSelection();
-                if ($isRangeSelection(selection)) {
-                    $wrapNodes(selection, () => $createTypographyNode(payload.value));
+                if ($isRangeSelection(selection) && payload.value.id && payload.value.tag === "p") {
```

Comment:
The thing I don't understand here... this is a TypographyPlugin, and we're checking if `payload.value.tag === "p"`. Why is this only working with `p` tags and not like h1?

Some comments are in order here I'd say. 🙂

Reply:
This typography plugin implements the event when the user will select one of the typography in the dropdown to apply to the currently selected node in this case paragraph (heading will be included as well).

**Additional explanation**:
The typography node itself is not flexible and modular as the separate heading and paragraph nodes in controlling the insert/updates. 
So we created `BaseHeadingNode` and `BaseParagraphNode` where all theme styles will be applied.